ForgeUnity/Assets/BeardedManStudios/Scripts/StandAlone/Objects/Float2.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace BeardedManStudios
+{
+	public partial struct Float2
```

Comment:
I'm wondering whether it would be better to use the Vector2, Vector3 and Vector4 from System.Numerics instead of defining Float2,3 and 4.

Reply:
I merely defined them so that they can be extended for different functionality as it's hard to see how people with specific use-cases would use them. I just defined Float 2, 3, and 4 because I have in my own projects as to not cause confusion. (It also, in some cases, causes you to have to write out UnityEngine.Vector2 or Numerics.Vector2, it's just cleaner to define Float2 and have them use separate names)

Reply:
What about using aliases?

Reply:
Well, the typing out the full reference wasn't the only issue (it was a pretty minimal one that could indeed be solved by using aliases). When I was looking into using System.Numerics I found a lot of issues with unity not playing nicely with it in a lot of scenarios. This was mostly before the .NET frameworks switch to 4.x and System.Numerics being included with that switch, but I have still seen a lot of issues pop up from devs attempting to use it with unity. We could switch over to the System.Numerics if it's deemed stable enough to use with Unity, but there is also the option of the end-user switching out the Float2-4 for the System.Numerics Vector2-4 if they want to use it, it should be set up in a way for the end-user to do that without any problems or extending the Float2-4 to suit unforeseeable needs.

Reply:
Fair enough.